src/Cake.Scripting/CodeGen/CakeScriptGenerator.cs

```diff
@@ -226,10 +226,41 @@ private DirectoryPath GetToolPath(DirectoryPath root)
 
         private DirectoryPath GetCakePath(DirectoryPath toolPath)
         {
+            DirectoryPath MakeAbsolute(FilePath filePath)
+            {
+                return filePath.GetDirectory().MakeAbsolute(_environment);
+            }
+
+            // Check for local cake in tools path
             var pattern = string.Concat(toolPath.FullPath, "/**/Cake.exe");
             var cakeCorePath = _globber.GetFiles(pattern).FirstOrDefault();
+            if (cakeCorePath != null)
+            {
+                return MakeAbsolute(cakeCorePath);
+            }
+
+            // Check for global cake-tool
+            var userprofile = _environment.GetEnvironmentVariable("USERPROFILE");
+            pattern = string.Concat(((DirectoryPath)userprofile).FullPath, "/.dotnet/tools/**/Cake.dll");
+            cakeCorePath = _globber.GetFiles(pattern).FirstOrDefault();
+            if (cakeCorePath != null)
+            {
+                return MakeAbsolute(cakeCorePath);
+            }
+
+            // Check for global chocolatey
+            var chocolateyInstall = _environment.GetEnvironmentVariable("ChocolateyInstall");
```

Comment:
I agree.  Don't think there is any need to check for Chocolatey install location.  If Cake is installed via Chocolatey, it will be on the path, nothing else required there.   Regarding your other question about homebrew, honestly, I am not sure.  I don't know how the internals work.  @devlead can you offer anything here?